src/textual/app.py

```diff
@@ -644,24 +644,27 @@ def save_screenshot(
         self,
         filename: str | None = None,
         path: str = "./",
-        time_format: str = "%Y%m%d %H%M%S %f",
+        time_format: str | None = None,
     ) -> str:
         """Save an SVG screenshot of the current screen.
 
         Args:
             filename (str | None, optional): Filename of SVG screenshot, or None to auto-generate
                 a filename with the date and time. Defaults to None.
             path (str, optional): Path to directory for output. Defaults to current working directory.
-            time_format (str, optional): Time format to use if filename is None. Defaults to "%Y-%m-%d %X %f".
+            time_format (str | None, optional): Date and time format to use if filename is None.
+                Defaults to ISO 8601 format. Colons are replaced with underscores.
 
         Returns:
             str: Filename of screenshot.
         """
         if filename is None:
-            svg_filename = (
-                f"{self.title.lower()} {datetime.now().strftime(time_format)}.svg"
-            )
-            for reserved in '<>:"/\\|?*':
+            if time_format is None:
+                dt = datetime.now().isoformat()
```

Comment:
We're not opposed to the change, although it doesn't make a whole lot of difference to how screenshots work. However, I see one small wrinkle that should be handled. This approach will result in more than one `.` in the resulting name. While permissible on all operating systems Textual runs on, it could be confusing for some people in some situations.  I think it would be better to result in an ISO8601-related format that doesn't run into that issue.

Reply:
Yes, I agree, it's a small change that doesn't make a whole lot of difference. Also, I have to admit that ISO8601 is not ideal for a file name -- the microsecond spec includes `:` and `.`. My misgivings with the current default are (a) I'm always nervous about spaces in a file name, and (b) the three groups of numbers -- date, time, and microseconds -- without any other separators look a little cryptic at first.

Do you think simply substituting the `.` used for microseconds with `_`, (eg., `demo_2023-01-07T22_52_31_512393.svg`), would be an improvement?

Reply:
Aye, that would make sense. Mostly I think we'd be expect people to pass in their own name anyway - the current form is only to help discourage a clash if people forget or choose not to. Swapping out the decimal point seems like the best approach here.